app/services/admin/onboarding_wizard.rb

```diff
@@ -0,0 +1,164 @@
+module Admin
+  class OnboardingWizard
+
+    MarketplaceSetupSteps = ::MarketplaceSetupSteps
+
+    KNOWN_STATUSES = [
+      :slogan_and_description, :cover_photo, :filter, :paypal, :listing, :invitation
+    ].to_set
+
+    EVENT_TYPES = [
+      :community_customizations_updated,
+      :community_updated,
+      :custom_field_created,
+      :paypal_preferences_updated,
+      :listing_created,
+      :invitation_created
+    ].to_set
+
+    SetupStatus = EntityUtils.define_builder(
+      [:community_id, :fixnum, :mandatory],
+      [:slogan_and_description, :bool, :mandatory],
+      [:cover_photo, :bool, :mandatory],
+      [:filter, :bool, :mandatory],
+      [:paypal, :bool, :mandatory],
+      [:listing, :bool, :mandatory],
+      [:invitation, :bool, :mandatory])
+
+    def initialize(community_id)
+      raise ArgumentError("Missing mandatory community_id") unless community_id
+      @community_id = community_id
+    end
+
+    # Get the status as a SetupStatus hash
+    def setup_status
+      to_setup_status(load_setup_steps(@community_id))
+    end
+
+    # Imperative shell. Process the given event_type with *args
+    # arguments. If the event leads to a state change apply it and
+    # return true. Otherwise return false.
+    def update_from_event(event_type, *args)
+      setup_status = to_setup_status(load_setup_steps(@community_id))
+      completed_status = process_event(event_type, setup_status, args)
+
+      if completed_status
+        update_completed(@community_id, completed_status)
+        true
+      else
+        false
+      end
+    end
+
+
+    private
+
+    def process_event(event_type, setup_status, args)
+      unless EVENT_TYPES.include?(event_type)
+        raise ArgumentError.new("Unkown event type: #{event_type}")
+      end
+
+      # Dispatch to event handler method of same name as event_type
+      method(event_type).call(setup_status, *args)
+    end
+
+
+    # Update events
+    #
+
+    def community_customizations_updated(setup_status, community_customizations)
+      if !setup_status[:slogan_and_description] &&
+         community_customizations.all? { |c| c.slogan.present? } &&
+         community_customizations.all? { |c| c.description.present? }
+        :slogan_and_description
+      end
+    end
+
+    def community_updated(setup_status, community)
+      if !setup_status[:cover_photo] &&
+         Maybe(community).map { |c| c.cover_photo_file_name.present? }.or_else(false)
```

Comment:
Not sure but just to be safe. I don't want the on boarding wizard to crash things easily since it's a non-critical experimental feature for now.